packages/remix-react/browser.tsx

```diff
@@ -324,6 +345,15 @@ export function RemixBrowser(_props: RemixBrowserProps): ReactElement {
             window.__remixRouteModules
           )
         : undefined,
+      async unstable_patchRoutesOnMiss(path, _, patch) {
+        if (!known404Paths.has(path)) {
+          await fetchAndApplyManifestPatches(
+            [path],
+            window.__remixManifest.version,
+            patch
+          );
+        }
+      },
```

Comment:
If we click a link that doesn't match, try to fetch that path from the server during the `loading` phase.  In an ideal scenario, this never runs because the pre-discovery logic all lands prior to link clicks.